### ISSUE TITLE
Error: Non-whitespace before first tag

### ISSUE THREAD

**Issue by michelem09** [isaacs/sax-js#138]
This is really weird, I have lot of Onix XML files I can handle right with sax-js but there are some, apparently identical, they throw this error:

error! [Error: Non-whitespace before first tag.
Line: 0
Column: 1
Char: ﻿]

There are no non-whitespaces at start of file.  I can't really understand.
Any ideas?

**Comment by ARIFCSE10** [comment]
which file need to edit

**Comment by CinKon** [comment]
Error is back in 1.2.1 :-/

**Comment by lsanchonq** [comment]
Same error .....

**Comment by jefabrah** [comment]
Was this resolved? I'm receiving the same error...

**Comment by EddieWen-Taiwan** [comment]
I got the same error.

**Comment by isaacs** [comment]
Can someone point me at an xml file that's generating this error?  It's impossible to fix without being able to reproduce.

**Comment by EddieWen-Taiwan** [comment]
This is my svg file
```
<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 36.223 36.223"><title>playbtn</title><path d="M31.957,16.943,5.689,1.777A1.422,1.422,0,0,0,3.555,3.009V33.341a1.422,1.422,0,0,0,2.133,1.232L31.957,19.407a1.422,1.422,0,0,0,0-2.463Z" style="fill:#fff"/></svg>
```

It's fine with [XML Validator](http://www.w3schools.com/xml/xml_validator.asp)

**Comment by isaacs** [comment]
This works fine for me. https://gist.github.com/isaacs/b737318fe60259e1ad1b5404f18d36bc

There's something else in the actual file you're parsing.  Upload it somewhere, please, or check it in a hex editor.  There are some bytes before the `<` char.

**Comment by dhritzkiv** [comment]
I believe this has something to do with a BOM character being inserted which Windows automatically prepends to the file. [See this blog post](http://www.multiasking.com/blog/xml2js-sax-js-non-whitespace-before-first-tag/)

**Comment by lafiza** [comment]
have the same error..

verified with XML Editor and as well as HEX editor. There are no BOM char prepended.

using xml2js - 0.4.17

http.get('../src/assets/admin-config.xml')
.subscribe(data => xml2js.parseString(data, function(err, result){
  console.log(err);
}));

xml format:
<?xml version="1.0" encoding="ISO-8859-1"?>
<config><section id="1"><title>Information Delivery</title><link id="1"><title>Communication Center</title><url>/messagecenter/MCHome</url><url-param>IBD,Role,NavInd,RR_Num,OffNum,account_id</url-param><hover-text>Messageson system outages, early closings, and product specific news.</hover-text><type>inline</type><popup-params/><frm-action>POST</frm-action></link></section></config>

**Comment by lafiza** [comment]
re-posting XML 
<config><section id="1"><title>Information Delivery</title><link id="1"><title>Communication Center</title><url>/messagecenter/MCHome</url><url-param>IBD,Role,NavInd,RR_Num,OffNum,account_id</url-param><hover-text>Messageson system outages, early closings, and product specific news.</hover-text><type>inline</type><popup-params/><frm-action>POST</frm-action></link></section></config>

**Comment by lafiza** [comment]
resending xml file :
[xml.txt](https://github.com/isaacs/sax-js/files/1152647/xml.txt)

**Comment by lafiza** [comment]
with angular-cli project., was able to achieve -

http.get('../../assets/admin-config.xml')
.subscribe(res => xml2js.parseString(res.text(), function(err, result){
  console.log(result);
}));

no error...

**Comment by dhritzkiv** [comment]
If I had to guess: perhaps the data in your first example was coming in as raw byte data, vs. being text that's utf8 encoded?

**Comment by pkmelee337** [comment]
I also was receiving this error and for me it was also the WIndows BOM character. Thanks @dhritzkiv for pointing this out. Resolved this issue by stripping this character first with the [gulp-stripbom package](https://www.npmjs.com/package/gulp-stripbom).

**Comment by tomoat** [comment]
How resolve ? same error

**Comment by ericketts** [comment]
yes this looks like a failure case due to the parser choking on a BOM. presently running into it on some SVG files sent by a vendor, looks like they were exporting with software that added a UTF-16 BOM, which crashes sax-js... 

if this failure is triggered to adhere to standards, but can be reasonably worked around (as implied by @michelem09 's workaround comment above), doesn't it make sense to deviate from the standards in the case that the illegal non-whitespace char is a byte order marker, for the sake of real world compatibility?
